### PR TITLE
Fixed Issue #25130:Issue with reorder when disbled reorder setting from admin

### PR DESCRIPTION
**Pull request by ghost**
### Fixed Issues (if relevant)

1. magento/magento2#25130: Issue with reorder when disabled reorder setting from admin

### Manual testing scenarios (*)

1. Place order from frontend as a customer or guest
2. Open order view page from frontend
3. Login into admin panel
4. Disable reorder option from admin. Navigate to Store -> Configuration-> Sales -> Reorder -> Allow Order and set this to **NO**
5. Go to opened order view page and click on reorder.
6. Now there will be an error message and you will not be able to reorder if setting is set to no from admin.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Sales/i18n/en_US.csv [hunk 1]
```diff
@@ -1,801 +1,801 @@
-"Credit Memos","Credit Memos"
```
Comment (swnsma) [inline app/code/Magento/Sales/i18n/en_US.csv]: Could you discard these changes and leave only required translate.

File: app/code/Magento/Sales/Controller/AbstractController/Reorder.php [hunk 2]
```diff
@@ -57,6 +57,14 @@ public function execute()
         /** @var \Magento\Framework\Controller\Result\Redirect $resultRedirect */
         $resultRedirect = $this->resultRedirectFactory->create();
 
+        $reorderHelper =  $this->_objectManager->get(\Magento\Sales\Helper\Reorder::class);
+        if (!$reorderHelper->canReorder($order->getEntityId())) {
+            $this->messageManager->addErrorMessage(__("Reorder is disabled."));
+            return $resultRedirect->setPath('checkout/cart');
+        }
+
+        $cart = $this->_objectManager->get(\Magento\Checkout\Model\Cart::class);
```
Comment (swnsma) [inline app/code/Magento/Sales/Controller/AbstractController/Reorder.php]: I do not think that we need to receive cart twice in the row :)

File: app/code/Magento/Sales/Controller/AbstractController/Reorder.php [hunk 3]
```diff
@@ -57,6 +57,14 @@ public function execute()
         /** @var \Magento\Framework\Controller\Result\Redirect $resultRedirect */
         $resultRedirect = $this->resultRedirectFactory->create();
 
+        $reorderHelper =  $this->_objectManager->get(\Magento\Sales\Helper\Reorder::class);
```
Comment (swnsma) [inline app/code/Magento/Sales/Controller/AbstractController/Reorder.php]: Can we move dependencies from Object Manager direct call to the class constructor?

File: app/code/Magento/Sales/Controller/AbstractController/Reorder.php [hunk 4]
```diff
@@ -57,6 +57,14 @@ public function execute()
...
+        if (!$reorderHelper->canReorder($order->getEntityId())) {
+            $this->messageManager->addErrorMessage(__("Reorder is disabled."));
```
Comment (swnsma) [inline app/code/Magento/Sales/Controller/AbstractController/Reorder.php]: May you change the phrase to the 'Reorder not available.' ?

**Review by swnsma — CHANGES_REQUESTED** [review]
Hi @rahulwebkul,
Thank you for your contribution!
Could you please check my comments?

**Comment by ghost** [comment]
@swnsma - I will check it and update it accordingly.

File: app/code/Magento/Sales/i18n/en_US.csv (hunk 1 above)
Comment (ghost) [reply app/code/Magento/Sales/i18n/en_US.csv]: @swnsma - I didn't get this. do you want me to remove this entry from csv?

File: app/code/Magento/Sales/Controller/AbstractController/Reorder.php (hunk 2 above)
Comment (ghost) [reply app/code/Magento/Sales/Controller/AbstractController/Reorder.php]: @swnsma - duplicate entry removed.

File: app/code/Magento/Sales/Controller/AbstractController/Reorder.php (hunk 3 above)
Comment (ghost) [reply app/code/Magento/Sales/Controller/AbstractController/Reorder.php]: @swnsma - Object manager was already used in file so i did not add any dependency.

File: app/code/Magento/Sales/Controller/AbstractController/Reorder.php (hunk 4 above)
Comment (ghost) [reply app/code/Magento/Sales/Controller/AbstractController/Reorder.php]: @swnsma - i have changed the phrase

File: app/code/Magento/Sales/i18n/en_US.csv (hunk 1 above)
Comment (swnsma) [reply app/code/Magento/Sales/i18n/en_US.csv]: @rahulwebkul,
In this file are 1,288 changed lines (644 lines have been changed).
I expected to see only 1 line added.

File: app/code/Magento/Sales/Controller/AbstractController/Reorder.php (hunk 3 above)
Comment (swnsma) [reply app/code/Magento/Sales/Controller/AbstractController/Reorder.php]: Yes, but please move `\Magento\Sales\Helper\Reorder` to the constructor.

**Review by swnsma — CHANGES_REQUESTED** [review]
Hi @rahulwebkul,
Thank you for your changes.
I have commented in conversations.

File: app/code/Magento/Sales/i18n/en_US.csv (hunk 1 above)
Comment (ghost) [reply app/code/Magento/Sales/i18n/en_US.csv]: @swnsma - csv file updated. please review the changes

File: app/code/Magento/Sales/Controller/AbstractController/Reorder.php (hunk 3 above)
Comment (ghost) [reply app/code/Magento/Sales/Controller/AbstractController/Reorder.php]: @swnsma - dependency added. Please review the changes

**Comment by ghost** [comment]
@swnsma -Please check the failed "Static Tests build".
It is showing "Possible useless method overriding detected".
The method is overridden in core magento. I did not do anything in it.

**Comment by ghost** [comment]
@swnsma - I have updated all the files and fixed the failed test case issue. Please preview the changes

**Comment by ghost** [comment]
@swnsma - Is there anything required from my end. Let me know so that i can update it accordingly.

**Comment by ghost** [comment]
@swnsma - Did you get the chance to review this PR. Please review and let me know if anything is pending from my end.

**Review by swnsma — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @swnsma, thank you for the review.
[ENGCOM-6167](https://jira.corp.magento.com/browse/ENGCOM-6167) has been created to process this Pull Request
:eight_spoked_asterisk: @swnsma, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Bravo** [comment]
Hi @swnsma, could You please add an appropriate label on auto-tests to this PR. Thank You.

**Comment by swnsma** [comment]
It seems to be possible to be covered by MFTF tests.

**Review by swnsma — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @swnsma, thank you for the review.
[ENGCOM-6167](https://jira.corp.magento.com/browse/ENGCOM-6167) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
Hello @rahulwebkul. Thank You for Your contribution. Please cover Your PR with the appropriate [auto-tests](https://github.com/magento/magento2/pull/25131#issuecomment-546666499). Thank You.

**Comment by VladimirZaets** [comment]
@rahulwebkul , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration

**Pull request closed without merging**